### PR TITLE
Adding Support for Sovereign clouds

### PR DESCRIPTION
**Pull request by aaronba**
Adding contribution to support #11 Issue:

### CODE REVIEW

**Comment by anweiss** [comment]
@SumiranAgg @N-Usha bump for review. Any chance we can get this merged? Thanks!

File: src/main.ts [hunk 1]
```diff
@@ -19,6 +19,12 @@ async function main() {
         azPath = await io.which("az", true);
         await executeAzCliCommand("--version");
 
+        let cloud = core.getInput('cloud', { required: false });
+        if (!cloud){
```
Comment (eaarora-ms) [inline src/main.ts:23]: Instead of saying if(!cloud) and leaving it empty, can we have 
if(cloud) {
// operation
}

**Comment by eaarora-ms** [comment]
@aaronba The PR looks good to me. Can you please make the changes I suggested and resolve the conflicts?
You can tag once the changes are done and I will approve the PR.
Thank you for your contributions, we really appreciate it.

File: src/main.ts (hunk 1 above)
Comment (aaronba) [reply src/main.ts]: @eaarora-ms , I made the suggested change.  Please review.

**Comment by eaarora-ms** [comment]
@N-Usha Can you please review it?

File: src/main.ts [hunk 2]
```diff
@@ -19,6 +19,12 @@ async function main() {
         azPath = await io.which("az", true);
         await executeAzCliCommand("--version");
 
+        let cloud = core.getInput('cloud', { required: false });
```
Comment (aksm-ms) [inline src/main.ts]: @aaronba Can you make let as const?

File: src/main.ts [hunk 3]
```diff
@@ -19,6 +19,12 @@ async function main() {
         azPath = await io.which("az", true);
         await executeAzCliCommand("--version");
 
+        let cloud = core.getInput('cloud', { required: false });
+        if (!cloud){
+        }else{
+            await executeAzCliCommand(`cloud set --name "${cloud}"`);
```
Comment (aksm-ms) [inline src/main.ts]: @aaronba please pass "true" as second argument to executeAzCliCommand. Refer other executeAzCliCommand calls

Comment (aaronba) [reply src/main.ts]: Done.  See latest commit.
and please take a look at ServicePrincipalLogin.ts as well for PowerShell case.
(code was added after initial PR)

**Comment by N-Usha** [comment]
Thanks @aaronba  for the PR. We are going to pick this enhancement over the next few weeks. For now closing this PR and we will surely update as we have the sovereign cloud support implemented.

**Pull request closed without merging**